### PR TITLE
Fix missed mouse out to hide status window (2)

### PR DESCRIPTION
**Pull request by ponchofiesta**
This adds the conf check for hiding window on mouse leave and checks if a context menu or drop down list is opened.

### CODE REVIEW

File: Nagstamon/QUI/__init__.py [hunk 1]
```diff
@@ -1620,9 +1622,11 @@ def hide_window_if_not_under_mouse(self):
...
         Hide window if it is under mouse
         """
-        mouse_x = QCursor.pos().x()
-        mouse_y = QCursor.pos().y()
-        if statuswindow.geometry().contains(mouse_x, mouse_y):
+        mouse_pos = QCursor.pos()
+
+        # Check mouse cursor over window and an opened context menu or dropdown list
+        if statuswindow.geometry().contains(mouse_pos.x(), mouse_pos.y()) \
+                or not qApp.activePopupWidget() is None:
```
Comment (HenriWahl) [inline Nagstamon/QUI/__init__.py]: Where does qApp come from? Shouldn't it be APP as defined in line 30?

Comment (ponchofiesta) [reply Nagstamon/QUI/__init__.py]: It is `QtWidget.qApp`. You have already imported everything from `QtWidget`. I assume `APP` is the same here.
https://doc.qt.io/qt-5/qapplication.html#qApp

Comment (HenriWahl) [reply Nagstamon/QUI/__init__.py]: Yes, APP is the current instance there.

**Comment by HenriWahl** [comment]
My shell's history lured me to do a forced git **push** instead a **pull** and so I overwrote you merged PR. Can you please request again?

**Comment by ponchofiesta** [comment]
You already remerged it. You could have rebased on my feature branch to get the correct commits again. Or cherry pick the missing commits.

**Comment by HenriWahl** [comment]
Yes, I included your changes as a patch file because they were not visible for me anymore after my forced push - either way, now they are in master.

**Comment by HenriWahl** [comment]
There is another bug or at least strange looking behaviour. The status window does not open or closes too fast when hovering the pointer in the floating statusbar in a Fedora Wayland session. Can you confirm this? Session is a default GNOME Wayland session.

**Comment by ponchofiesta** [comment]
Sorry, can't verify. I don't have any of this software installed.

**Comment by HenriWahl** [comment]
I added a simple Wayland detection and disabled your patch in case of running in Wayland. Maybe there might be a more sophisticated solution but as there are problems with Wayland in general I think it doesn't matter.

**Comment by HenriWahl** [comment]
Me again.

Could it be that the problem you try to solve is caused by https://github.com/HenriWahl/Nagstamon/blob/master/Nagstamon/QUI/__init__.py#L1665?

Apparently your fix causes new trouble in Windows, making the floating statusbar jump along the screen from time to time...

**Pull request merged**